examples/nextjs-with-typescript/pages/MuxPlayer.tsx

```diff
@@ -293,477 +235,371 @@ const getControlCustomizationCSSVars = (state) => {
     .map(([k]) => k);
 };
 
-type Props = { location?: Pick<Location, 'origin' | 'pathname'> };
-
-const getUrl = ({ req, resolvedUrl }) => {
-  const { headers } = req;
-  const refererUrl = headers.referer && new URL(headers.referer);
-  const baseUrlHost = headers.host.toLowerCase();
-  const refererHost = refererUrl?.host?.toLowerCase();
-
-
-  if (refererHost === baseUrlHost && headers["sec-fetch-site"] === "same-origin") {
-    return new URL(refererUrl?.origin ? refererUrl.origin + resolvedUrl : './');
-  }
-
-  const startsLocal = baseUrlHost.startsWith('localhost') || baseUrlHost.startsWith('127.') || baseUrlHost.startsWith('192.');
-  const protocol = startsLocal ? 'http:' : 'https:';
-  return new URL(`${protocol}//${baseUrlHost}${resolvedUrl}`);
-};
-
-export const getServerSideProps: GetServerSideProps<Props> = async context => {
-  const { origin, pathname }: Pick<Location, 'origin' | 'pathname'> = getUrl(context);
-  const location = { origin, pathname };
-  return ({ props: { location } })
-};
-
 const MaxResolutionValues = Object.values(MaxResolution);
 const MinResolutionValues = Object.values(MinResolution);
 const RenditionOrderValues = Object.values(RenditionOrder);
 
+export const getServerSideProps = getLocationServerSideProps;
+
+type Props = LocationProps;
+
 function MuxPlayerPage({ location }: Props) {
-  const router = useRouter();
-  const mediaElRef = useRef(null);
+  // const mediaElRef = useRef<MuxPlayerElement>(null);
   const [mediaAssets, _setMediaAssets] = useState(mediaAssetsJSON);
   const [selectedAsset, setSelectedAsset] = useState(undefined);
-  const [state, dispatch] = useReducer(reducer, toInitialState(selectedAsset, mediaAssets, router.query));
-  useEffect(() => {
-    if (!router.isReady) return;
-    dispatch(updateProps(toInitialState(selectedAsset, mediaAssets, router.query)))
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [router.query, router.isReady]);
+  /** @TODO fix typing complexities here (CJP) */
+  // @ts-ignore
```

Comment:
There were some type complexities that would have also required some non-trivial type rearchitecting (e.g. adding some `infer` logic etc.). Hoping to either circle back around to this or that some of those complexities will fall away with my proposed reworking of initial state crud, as mentioned above.